policy-controller/src/admission.rs

```diff
@@ -1,24 +1,41 @@
 use crate::api;
-use anyhow::{anyhow, bail, Context, Result};
+use anyhow::{anyhow, bail, Result};
 use api::policy::ServerSpec;
 use futures::future;
 use hyper::{body::Buf, http, Body, Request, Response};
-use kube::{api::Api, core::DynamicObject, ResourceExt};
-use std::convert::TryInto;
+use kube::{core::DynamicObject, Api, ResourceExt};
 use std::task;
+use thiserror::Error;
 use tracing::{debug, info, warn};
 
 #[derive(Clone)]
 pub struct Service {
     pub client: kube::Client,
 }
 
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("failed to read request body: {0}")]
+    Request(
+        #[from]
+        #[source]
```

Comment:
nit: i don't believe we need both of these; per the `thiserror` README, 
>   The `#[from]` attribute always implies that the same field is `#[source]`, so
  you don't ever need to specify both attributes.
```suggestion
        #[from]
```